RundeckAutomation/Functions/New-RundeckSession.ps1

```diff
@@ -0,0 +1,64 @@
+function New-RundeckSession
+{
+    <#
+    .SYNOPSIS
+    Creates a connection to Rundeck instance.
+
+    .DESCRIPTION
+    The `New-RundeckSession` function creates a connection to a Rundeck instance. Pass the URI to the instance to the `Uri` parameter. Pass the credentials to use to authenticate to the `Credential` parameter. 
+
+    The function generates a token for further access to the Rundeck API using the credentials by making a request to the `/api/v2/api-token-auth` endpoint.
+
+    .EXAMPLE
+    New-RundeckSession -Uri 'https://Rundeck.example.com' -Credential $me
+
+    Demonstrates how to connect to an instance of Rundeck. In this case, the connection is to the instance at `https://rundeck.example.com` using the credentials in the `$me` variable.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [uri]
+        # The URI to the Rundeck instance to connect to.
+        $Uri,
+
+        [Parameter(Mandatory, ParameterSetName = 'username')]
+        [pscredential]
+        # The credentials to use.
+        $Credential,
+
+        [Parameter(Mandatory, ParameterSetName = 'apitoken')]
+        [string]
+        # The credentials to use.
+        $ApiToken,

```

Comment:
Can you allow both a credential and an API token?

Another advantage to returning a session object instead of using a global object is users can mix calls to endpoints that require/use credentials with calls to endpoints that require/use the API token without needing to call `Set-RundeckSession` again. Each function would tell `Invoke-RundeckRestMethod` to use credentials or an API key.

Reply:
All calls should/have been able to support both.  I had a syntax error that led me to believe otherwise for one of the endpoints.  Thinking on a compromise.  I like setting it at the session level (like VMWare's PowerCLI supports), but see the value in being able to specify.